### ISSUE TITLE
Scaling up local storage

### ISSUE THREAD

**Issue by NicolasT** [scality/metalk8s#175]
From ZENKO-792:

> How can we scale storage? Metal-k8s deploys on drives using: metal_k8s_lvm defined in kube-node.yml as stated in https://metal-k8s.readthedocs.io/en/latest/usage/quickstart.html. Need the functionality of adding more storage on an already deployed metal-k8s and zenko cluster(probably with a playbook).

**Comment by NicolasT** [comment]
Note: the description above refers to pre-0.2 configuration which changes with #94 and #153.

Extending storage can be achieved in 2 ways:

- Add new disks to the set of disks backing a previously deployed VG, and re-run the storage deployment
- Add a new storage pool (i.e. a new LVM VG with new disks, define LVs,...), and re-run the storage deployment

Re-running the storage deployment can be achieved using one of the following ways (and maybe others):

- Re-run the whole `deploy` playbook
- Re-run the `deploy` playbook with `--tags storage` (note: this is not necessarily a stable 'API' yet)
- Run the `storage-pre` and `storage-post` playbooks (which means we may want to have another playbook, `storage`, which basically runs those two)

The above is currently not a tested scenario, so we should add this to the test-plan at some point and automate.

**Comment by anurag4DSB** [comment]
Hello Nicolas,

Thank you for responding to this.
So is this what you are suggesting:
ansible-playbook -i inventory/quickstart-cluster -b playbooks/deploy.yml
ansible-playbook -i inventory/quickstart-cluster -b playbooks/storage-pre.yml
ansible-playbook -i inventory/quickstart-cluster -b playbooks/storage-post.yml

Can you also give an example of adding new metal_k8s_lvm files?

**Comment by NicolasT** [comment]
No, not exactly (although it shouldn't do any harm): the `deploy` playbook already runs `storage-pre` and `storage-post`. The list of options I presented was

> one of the following ways

W.r.t. adding new VGs, this is currently being overhauled completely so we'll work on documenting this feature once (or actually after, in this case) the changes land (cfr. #153).

**Comment by NicolasT** [comment]
This is being documented in #400 / #385.

**Comment by thomasdanan** [comment]
#1997
